addons/binding/org.openhab.binding.rfxcom/src/main/java/org/openhab/binding/rfxcom/handler/RFXComHandler.java

```diff
@@ -160,131 +151,34 @@ public void onDeviceMessageReceived(ThingUID bridge, RFXComMessage message) {
-                                        break;
-                                    case STATUS:
-                                        updateState(CHANNEL_STATUS, message.convertToState(valueSelector));
-                                        break;
-                                    case TEMPERATURE:
-                                        updateState(CHANNEL_TEMPERATURE, message.convertToState(valueSelector));
-                                        break;
-                                    case CHILL_TEMPERATURE:
-                                        updateState(CHANNEL_CHILL_TEMPERATURE, message.convertToState(valueSelector));
-                                        break;
-                                    case TOTAL_AMP_HOUR:
-                                        updateState(CHANNEL_TOTAL_AMP_HOUR, message.convertToState(valueSelector));
-                                        break;
-                                    case TOTAL_USAGE:
-                                        updateState(CHANNEL_TOTAL_USAGE, message.convertToState(valueSelector));
-                                        break;
-                                    case UV:
-                                        updateState(CHANNEL_UV, message.convertToState(valueSelector));
-                                        break;
-                                    case VOLTAGE:
-                                        updateState(CHANNEL_VOLTAGE, message.convertToState(valueSelector));
-                                        break;
-                                    case WIND_DIRECTION:
-                                        updateState(CHANNEL_WIND_DIRECTION, message.convertToState(valueSelector));
-                                        break;
-                                    case AVG_WIND_SPEED:
-                                        updateState(CHANNEL_AVG_WIND_SPEED, message.convertToState(valueSelector));
-                                        break;
-                                    case WIND_SPEED:
-                                        updateState(CHANNEL_WIND_SPEED, message.convertToState(valueSelector));
-                                        break;
-                                    default:
-                                        logger.debug("Unsupported value selector '{}'", valueSelector);
-                                        break;
-                                }
-                            } catch (RFXComException e) {
-                                logger.trace("{} does not handle {}", valueSelector, message);
+                    for (Channel channel : getThing().getChannels()) {
+                        String channelId = channel.getUID().getId();
+
+                        try {
+                            switch (channelId) {
+                                case CHANNEL_BATTERY_LEVEL:
+                                    updateState(channelId, convertBatteryLevelToSystemWideLevel(
```

Comment:
Given that on all places we already go to troubles of converting the integer value to a `DecimalType` we could as well call the conversion from there, now we first unpack the `DecimalType` do some math and repack it to a `DecimalType`. WDYT?